### ISSUE TITLE
Test-TCP-2.22_01-TRIAL: On Windows 8.1, t/HTTP-Server-PSGI/harakiri.t (Plack 1.0047) seems to hang

### ISSUE THREAD

**Issue by twata1** [tokuhirom/Test-TCP#95]
Hello,

When I try to install Plack 1.0047 under Strawberry Perl 5.16.3 on Windows 8.1, It seems to me that Test-TCP hangs as follows.

```
cpan Plack
```

(snip)
```
Running make test
Skip blib\lib\auto\share\dist\Plack\baybridge.jpg (unchanged)
Skip blib\lib\auto\share\dist\Plack\face.jpg (unchanged)
C:\Strawberry\perl\bin\perl.exe "-MExtUtils::Command::MM" "-e" "test_harness(0, 'blib\lib', 'blib\arch')" t/*.t t/HTTP-M
essage-PSGI/*.t t/HTTP-Server-PSGI/*.t t/Plack-Builder/*.t t/Plack-HTTPParser-PP/*.t t/Plack-Handler/*.t t/Plack-Loader/
*.t t/Plack-MIME/*.t t/Plack-Middleware/*.t t/Plack-Middleware/cascade/*.t t/Plack-Middleware/recursive/*.t t/Plack-Midd
leware/stacktrace/*.t t/Plack-Request/*.t t/Plack-Response/*.t t/Plack-Runner/*.t t/Plack-TempBuffer/*.t t/Plack-Test/*.
t t/Plack-Util/*.t
t/author-pod-syntax.t ................................ skipped: these tests are for testing by the author
t/HTTP-Message-PSGI/content_length.t ................. ok
t/HTTP-Message-PSGI/empty_streamed_response.t ........ ok
t/HTTP-Message-PSGI/host.t ........................... ok
t/HTTP-Message-PSGI/path_info.t ...................... ok
t/HTTP-Message-PSGI/unknown_response.t ............... ok
t/HTTP-Message-PSGI/utf8_req.t ....................... ok
[Test::TCP] Child process does not block(PID: -7164, PPID: 5528) at C:\home\pianokey\perl5\lib\perl5/Test/TCP.pm line 103.
```
(hangs)

```
[Environment]

Windows 8.1 (64bit)
Strawberry Perl 5.16.3 (64bit)

Apache::LogFormat::Compiler   0.36
Cookie::Baker                 0.11
Devel::StackTrace             2.04
Devel::StackTrace::AsHTML     0.15
File::ShareDir                1.03
Filesys::Notify::Simple       0.14
Hash::MultiValue              0.16
HTTP::Entity::Parser          0.22
HTTP::Headers::Fast           0.22
HTTP::Message                 6.18
HTTP::Tiny                    0.076
parent                        0.225
Pod::Usage                    1.61
Stream::Buffered              0.03
Try::Tiny                     0.12
URI                           1.60
WWW::Form::UrlEncoded         0.26

ExtUtils::MakeMaker           6.64
Test::More                    1.302170
Test::Requires                0.06
File::ShareDir::Install       0.13
```

Thank you,

**Comment by nponeccop** [comment]
1. It would be helpful to see a minimal example.
2. We have #74 which doesn't happen every time. Can you run the tests several times and see if it's random or happens every time?
3. Does it happen with older Test::TCP versions? Note that some older versions can't even be installed on Windows. Try 2.19

**Comment by twata1** [comment]
1. Sorry. It will take some time to create a minimal example.

2. It happens every time.
   (I tried to `cpan Plack` 4 times with the same result.)

3. It happens with Test::TCP 2.19 too.
   (I tried to `cpan Plack` 4 times with the same result.)

**Comment by twata1** [comment]
I'm sorry for troubling you.
I remembered my own comment https://github.com/plack/Plack/issues/418#issuecomment-429607642 .

Maybe it seems to happen on Windows at least if the version of LWP is old.

I upgraded Net::HTTP from version 6.06 to 6.19 and `cpan Plack` .
--> hangs

I upgraded LWP from version 6.04 to 6.43 and `cpan Plack` .
--> the issue disappeared

I close the ticket.

Thank you for your trouble.
